include/amber/amber.h

```diff
@@ -52,6 +52,17 @@ struct BufferInfo {
   std::vector<Value> values;
 };
 
+/// Delegate class for various hook functions
+class Delegate {
+ public:
+  virtual ~Delegate() = default;
```

Comment:
This will fail on the bots due to a warning being treated as error. You need to put the definition into the src/amber.cc file. So just virtual ~Delegate(); here and later Just Delegate::~Delegate() = default; in the .cc

Reply:
I tried what you said before, but it leads to the following error at link time (Linux gcc):

`samples/CMakeFiles/amber.dir/amber.cc.o:(.data.rel.ro+0x10): undefined reference to `typeinfo for amber::Delegate'`

Chatting with Ally and Ben, it turned out the issue was that `-fno-rtti` is used for all library code, but not the sample.

See the change on the sample CMakeLists.txt which fixes this.